src/hacbs/components/ApplicationDetails/HacbsApplicationDetails.tsx

```diff
@@ -0,0 +1,120 @@
+import * as React from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useK8sWatchResource } from '@openshift/dynamic-plugin-sdk-utils';
+import { Bullseye, Spinner } from '@patternfly/react-core';
+import { NamespaceContext } from '../../../components/NamespacedPage/NamespacedPage';
+import { ApplicationGroupVersionKind } from '../../../models';
+import { ApplicationKind } from '../../../types';
+import DetailsPage from './DetailsPage';
+import ApplicationOverviewTab from './tabs/ApplicationOverviewTab';
+import CommitsTab from './tabs/CommitsTab';
+import ComponentsTab from './tabs/ComponentsTab';
+import EnvironmentsTab from './tabs/EnvironmentsTab';
+import PipelineRunsTab from './tabs/PipelineRunsTab';
+
+type HacbsApplicationDetailsProps = {
+  applicationName: string;
+};
+
+const HacbsApplicationDetails: React.FC<HacbsApplicationDetailsProps> = ({ applicationName }) => {
+  const { namespace } = React.useContext(NamespaceContext);
+  const navigate = useNavigate();
+  const [application, applicationsLoaded] = useK8sWatchResource<ApplicationKind>({
+    groupVersionKind: ApplicationGroupVersionKind,
+    name: applicationName,
+    namespace,
+  });
+  const appDisplayName = application?.spec?.displayName || '';
+
+  const loading = (
+    <Bullseye>
+      <Spinner data-test="spinner" />
+    </Bullseye>
+  );
+
+  if (!applicationsLoaded) {
+    return loading;
+  }
+
+  return (
+    <React.Fragment>
+      <DetailsPage
+        breadcrumbs={[
+          { path: '/app-studio/applications', name: 'Applications' },
+          {
+            path: `/app-studio/applications/:${applicationName}`,
+            name: appDisplayName,
+          },
+        ]}
+        title={appDisplayName}
+        actions={[
+          {
+            key: 'add-component',
+            label: 'Add Component',
+            onClick: () => {
+              navigate(`/app-studio/import?application=${applicationName}`);
+            },
+          },
+          {
+            key: 'add-environment',
+            label: 'Add Environement',
+            href: '',
```

Comment:
Always good to have an href so that a user can copy the link or open in a new tab.